src/neuroconv/tools/nwb_helpers/_models/_hdf5_models.py

```diff
@@ -74,3 +75,18 @@ def get_data_io_keyword_arguments(self) -> Dict[str, Any]:
             compression_bundle = dict(compression=self.compression_method, compression_opts=self.compression_options)
 
         return dict(chunks=self.chunk_shape, **compression_bundle)
+
+
+class HDF5BackendConfiguration(BackendConfiguration):
```

Comment:
Can you get away with just using instances of `BackendConfiguration`, why do you need a class for `HDF5BackendConfiguration` and another for `ZarrBackendConfiguration`?

Reply:
Discussed in person: summary, it would require us to use a root validator for the common parent, which is a bit hackier

This works cleaner at the price of some redundancy